gmcs/feature_type_use.py

```diff
@@ -0,0 +1,7 @@
+# Attempt to flag specific features which were added to the grammar by a component.
+# One such feature is INIT; it is not needed in all grammars but can be added by the Word Order library,
+# the clausal complements library, the clausal modifiers library, and the Wh-questions library
+# (and possibly other libraries).
+
+USED_FEATURES = {'INIT':False}
+USED_TYPES = {'qdet':False,'qpro':False,'qadv':False,'qverb':False}
```

Comment:
I wanted to start collecting some statistics about what features are used how. I did not think about it much in terms of doing it over multiple grammars simultaneously although it would of course be interesting. But even within one grammar is interesting enough.

Of course I only started it and didn't really do anything interesting with these dicts yet.

Reply:
I think T.J. means that in a multithreaded (not multiprocess) use of the Matrix (which we do not currently do but in the future who knows) this global might be modified by different customizations in the same Python process, leading to incorrect results. But, as a side-point, if these are intended to be modified they should not be ALL_CAPS as that indicates they are constants.

Reply:
(another reason to not use multithreading, as far as I am concerned :) ).

Reply:
We're not using multithreading

Reply:
I know :). Good :). (Sorry, I am just messing with ya at this point)